### ISSUE TITLE
'timeoutController.clear' is undefined in Safari

### ISSUE THREAD

**Issue by icidasset** [libp2p/js-libp2p#568]
Hey folks,

I'm getting the following error in Safari v13.0.4 (MacOS v10.14.6)
```
TypeError: timeoutController.clear is not a function. (In 'timeoutController.clear()', 'timeoutController.clear' is undefined)
```

Which comes from:
https://github.com/libp2p/js-libp2p/blob/a317a8b0118ea03b76cdc430c9e28e2451b84532/src/dialer/index.js#L160

Using js-ipfs v0.41.1

No issues in Firefox or Chrome.
Do I need a polyfill for something? 🤔

**Comment by jacobheun** [comment]
Is there an additional stack trace with the error? Is js-ipfs just running in the background or does the error occur after specific actions?

It shouldn't need a polyfill as Safari supports `AbortController`. It's odd that the method is undefined since we don't modify the controller.

**Comment by icidasset** [comment]
<img width="943" alt="Screenshot 2020-02-21 at 21 19 46" src="https://user-images.githubusercontent.com/296665/75068695-f4d8fe80-54ef-11ea-978c-ded96e156304.png">
Where `(anonymous function) — ipfs.js:154653` points to:
<img width="504" alt="Screenshot 2020-02-21 at 21 54 24" src="https://user-images.githubusercontent.com/296665/75071017-bf82df80-54f4-11ea-9f8c-d0ea5b5f3843.png">

Happens when connecting to a peer using `swarm.connect` (I'm not passing any peers to `create()`). I'm using a websocket connection.

Trying to find out more, but yeah, it's not making much sense 👀

**Comment by jacobheun** [comment]
How are you building ipfs for the browser (Babel, webpack, etc)? I am wondering if this is related to https://github.com/jacobheun/timeout-abort-controller/issues/2, in that built in classes (AbortController) can have issues being extended when being transpiled.

**Comment by icidasset** [comment]
I'm using the [`unpkg`](https://unpkg.com/ipfs@0.41.2/dist/index.js) distribution, which is a webpack build by the looks of it. It does not use `super.call` which you talked about in the other issue.

**Comment by jacobheun** [comment]
It looks like these are actually related but for different reasons. In Safari it doesn't look like you can extend built in classes. The TimeoutController instance ends up just being an AbortController, so it doesn't have the `clear` method.

I'll look at compiling or updating the TimeoutController package to use prototypal inheritance to resolve these two issues.

**Comment by autonome** [comment]
Would love to see webcompat issues like these fixed. @jacobheun are you able to look anytime soon or should we see if someone else can pick it up?

**Comment by jacobheun** [comment]
Currently looking into this.

**Comment by oed** [comment]
Do you have any update or pointers for this issue @jacobheun?
It's currently blocking us from updating from `ipfs@0.40` :)

**Comment by jacobheun** [comment]
Sorry for the delay here, I just released a patch for `timeout-abort-controller@1.1.1` that patches the issue with Safari. @oed can you verify `timeout-abort-controller@1.1.1` fixes the issue for you?

**Comment by oed** [comment]
Thanks @jacobheun this solved the issue!
